### PR TITLE
Add annotation example in CircleCI plugin README

### LINKED ISSUE #3079: Circle CI plugin requesting from wrong project
I have a project which lives in CircleCI at `https://app.circleci.com/pipelines/github/RoadieHQ/sample-service`.

The annotation for the component that uses this build pipeline is [here](https://github.com/RoadieHQ/sample-service/blob/master/catalog-info-2.yaml).

```
  annotations:
    circleci.com/project-slug: RoadieHQ/sample-service
    rollbar.com/project-slug: roadie/sample-project
```

I have an auth token configured in `CIRCLECI_AUTH_TOKEN` (this backstage instance is running on kubernetes)

```
» kubectl -n demo exec backstage-backend -- printenv | grep CIRCLE
CIRCLECI_AUTH_TOKEN=[redacted]
```

When I visit [the component in Backstage](https://demo.roadie.so/catalog/Component/default:sample-service-2/ci-cd), I get a "Request failed with status code 404".

The request to fetch the builds from Circle CI (via the proxy), looks like this:

```
https://demo.roadie.so/api/proxy/circleci/api/project/bitbucket/sample-service/undefined?limit=5&offset=0&circle-token=
```

It seems to be assuming that this component is hosted on Bitbucket, which is wrong. Also, it shouldn't expect the token to be in the client side request.

Totally willing to accept that I've done something wrong when setting this up of course.

### PR DESCRIPTION
**Pull request by dtuite**
Closes #3079

### CODE REVIEW

**Review by freben — APPROVED** [review]
Nice! This one should also be in https://github.com/spotify/backstage/blob/master/docs/features/software-catalog/well-known-annotations.md i guess

File: plugins/circleci/README.md [hunk 1]
```diff
@@ -63,6 +63,19 @@ proxy:
 5. Get and provide `CIRCLECI_AUTH_TOKEN` as env variable (https://circleci.com/docs/api/#add-an-api-token)
 6. Add `circleci.com/project-slug` annotation to your component-info.yaml file in format <git-provider>/<owner>/<project> (https://backstage.io/docs/architecture-decisions/adrs-adr002#format)
 
+```
```
Comment (freben) [inline plugins/circleci/README.md:66]: oh hey nitpick - it's nice to start this with ` ```yaml ` so you get syntax highlighting

Comment (dtuite) [reply plugins/circleci/README.md]: Fixed. It's two commits now but I assume you can squash them when you merge?

Comment (freben) [reply plugins/circleci/README.md]: Yupyup, no problem

**Comment by dtuite** [comment]
@freben Your wish is my command! https://github.com/spotify/backstage/pull/3082

**Pull request merged**